### ISSUE TITLE
Gradients with respect to a TensorArray

### ISSUE THREAD

**Issue by maharjun** [tensorflow/tensorflow#23074]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using):
1.10.0
- Are you willing to contribute it (Yes/No):
No (not enough expertise)

**Describe the feature and the current behavior/state.**
It appears that one cannot use the `tf.gradients` function for taking derivatives with respect to TensorArrays. I would like there to be the feature to return a TensorArray with the relevant gradients

**Will this change the current api? How?**
Slightly but in a non-breaking manner (hopefully). It would be an addition to the capabilities of `tf.gradients`

**Who will benefit with this feature?**
This is a feature that is crucial if we wish to take derivatives with respect to intermediate states in a recurrent neural network.

Also related: issue #5412 (currently closed due to no response)

**Any Other info.**

**Comment by ymodak** [comment]
Can you please try and put one example that can explain the expected functionality that you wish to see?
It can be a small script that shows current behavior and you can comment what is your desired behavior from the function. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by ymodak** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
